packages/go/index.ts

```diff
@@ -94,332 +99,454 @@ export async function build({
+    if (renamedEntrypoint) {
+      await move(join(workPath, entrypoint), join(workPath, renamedEntrypoint));
+      undoFileActions.push({
+        to: join(workPath, entrypoint),
+        from: join(workPath, renamedEntrypoint),
+      });
+      entrypoint = renamedEntrypoint;
     }
-    analyzed = await getAnalyzedEntrypoint(
-      workPath,
-      downloadedFiles[entrypoint].fsPath,
-      goModAbsPathDir
-    );
-  } catch (err) {
-    console.log(`Failed to parse AST for "${entrypoint}"`);
-    throw err;
-  }
 
-  if (!analyzed) {
-    const err = new Error(
-      `Could not find an exported function in "${entrypoint}"
-Learn more: https://vercel.com/docs/runtimes#official-runtimes/go
-      `
-    );
-    console.log(err.message);
-    throw err;
-  }
+    const entrypointAbsolute = join(workPath, entrypoint);
+    const entrypointArr = entrypoint.split(sep);
 
-  const parsedAnalyzed = JSON.parse(analyzed) as Analyzed;
-
-  // find `go.mod` in downloadedFiles
-  const entrypointDirname = dirname(downloadedFiles[entrypoint].fsPath);
-  let isGoModExist = false;
-  let goModPath = '';
-  let isGoModInRootDir = false;
-  for (const file of Object.keys(downloadedFiles)) {
-    const { fsPath } = downloadedFiles[file];
-    const fileDirname = dirname(fsPath);
-    if (file === 'go.mod') {
-      isGoModExist = true;
-      isGoModInRootDir = true;
-      goModPath = fileDirname;
-    } else if (file.endsWith('go.mod')) {
-      if (entrypointDirname === fileDirname) {
-        isGoModExist = true;
-        goModPath = fileDirname;
-        debug(`Found file dirname equals entrypoint dirname: ${fileDirname}`);
-        break;
+    debug(`Parsing AST for "${entrypoint}"`);
+    let analyzed: string;
+    try {
+      const goModAbsPath = await findGoModPath(workPath);
+      if (goModAbsPath) {
+        debug(`Found ${goModAbsPath}"`);
       }
 
-      if (!isGoModInRootDir && config.zeroConfig && file === 'api/go.mod') {
-        // We didn't find `/go.mod` but we found `/api/go.mod` so move it to the root
+      analyzed = await getAnalyzedEntrypoint(
+        workPath,
+        entrypointAbsolute,
+        dirname(goModAbsPath)
+      );
+    } catch (err) {
+      console.log(`Failed to parse AST for "${entrypoint}"`);
+      throw err;
+    }
+
+    if (!analyzed) {
+      const err = new Error(
+        `Could not find an exported function in "${entrypoint}"
+  Learn more: https://vercel.com/docs/runtimes#official-runtimes/go
+        `
+      );
+      console.log(err.message);
+      throw err;
+    }
+
+    const parsedAnalyzed = JSON.parse(analyzed) as Analyzed;
+
+    // find `go.mod` in downloadedFiles
+    const entrypointDirname = dirname(entrypointAbsolute);
+    let isGoModExist = false;
+    let goModPath = '';
+    let isGoModInRootDir = false;
+
+    const modFileRefs = await glob('**/*.mod', workPath);
+    const modFiles = Object.keys(modFileRefs);
+
+    for (const file of modFiles) {
```

Comment:
Is `file` a relative or absolute path? It should probably use the name `fsPath` if its absolute

Reply:
It's relative. I didn't realize the convention was to use `fsPath` for absolute values. I'll do that going forward!